Kernel/Storage/AHCIPort.cpp

```diff
@@ -305,32 +318,63 @@ void AHCIPort::set_sleep_state() const
     m_port_registers.cmd = (m_port_registers.cmd & 0x0ffffff) | (0b1000 << 28);
 }
 
-void AHCIPort::start_request(AsyncBlockDeviceRequest& request)
+Optional<size_t> AHCIPort::calculate_descriptors_count(size_t block_count) const
```

Comment:
This returns `Optional<size_t>` but never fails? In what situation should this fail?

Reply:
The idea was to not return anything with the `Optional` container if we don't have enough DMA pages for a request, but for now I prefer to simply assert on that, so for now we can return `size_t` :)